src/inmanta/data/dataview.py

```diff
@@ -1294,11 +1293,19 @@ def get_base_query(self) -> SimpleQueryBuilder:
                     dr.environment,
                     dr.discovered_resource_id,
                     dr.values,
-                    (rps.resource_id IS NOT NULL)  AS managed
+                    (rps_1.resource_id IS NOT NULL)  AS managed,
+                    rps_2.resource_id as discovery_resource_id
 
                 FROM {data.DiscoveredResource.table_name()} as dr
-                LEFT JOIN {data.ResourcePersistentState.table_name()} rps
-                ON dr.environment=rps.environment AND dr.discovered_resource_id = rps.resource_id
+
+                -- Retrieve the corresponding managed resource id (if any)
+                LEFT JOIN {data.ResourcePersistentState.table_name()} rps_1
+                ON dr.environment=rps_1.environment AND dr.discovered_resource_id = rps_1.resource_id
+
+                -- Retrieve the id of the resource responsible for discovering this resource
+                LEFT JOIN {data.ResourcePersistentState.table_name()} rps_2
+                ON dr.environment=rps_2.environment AND dr.discovery_resource_id = rps_2.resource_id
```

Comment:
What's the purpose of this join? We join on resource id, then select resource id?

Reply:
This is to make sure that the `discovery_resource_id` corresponds to a proper resource that exists in the db.

Reply:
Ah right, because we link to it. Ok, fine by me.